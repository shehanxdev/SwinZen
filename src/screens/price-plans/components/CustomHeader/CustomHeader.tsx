import React from 'react';
import { GestureResponderEvent, Platform, TouchableOpacity, View } from 'react-native';

import { BackIcon, Text } from '@sz/components';
import { tw } from '@sz/config';
import { Color, TextVariant, headerTopMargin } from '@sz/constants';

export interface CustomHeaderProps {
  testID?: string;
  title: string;
  onBackPress: (event: GestureResponderEvent) => void;
}

export function CustomHeader({ testID = 'CustomHeaderTestID', title, onBackPress }: CustomHeaderProps) {
  const customMargin = Platform.OS === 'ios' ? 0 : headerTopMargin;

  return (
    <View testID={testID} style={tw`flex-row w-screen justify-between mt-[${customMargin}px]`}>
      <TouchableOpacity style={tw`px-4`} onPress={onBackPress}>
        <BackIcon />
      </TouchableOpacity>
      <Text variant={TextVariant.SubTitle2SemiBold} color={Color.Neutral.Sz100}>
        {title}
      </Text>
      <View style={tw`w-12`} />
    </View>
  );
}
