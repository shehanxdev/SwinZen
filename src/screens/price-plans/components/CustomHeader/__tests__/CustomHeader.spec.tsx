import { render } from '@testing-library/react-native';
import * as React from 'react';

import { CustomHeader } from './../CustomHeader';

describe('CustomHeader Component', () => {
  const testID = 'CustomHeaderTestID';
  const title = 'testTitle';
  const mockOnBackPressFunction = jest.fn();

  const getRenderedComponent = () =>
    render(<CustomHeader testID={testID} title={title} onBackPress={mockOnBackPressFunction} />);

  it(`should render CustomHeader component correctly`, () => {
    const rendered = getRenderedComponent();
    const renderedTree = rendered.toJSON();
    expect(renderedTree).toMatchSnapshot();
  });

  it('should find the CustomHeader component via testID', () => {
    const { getByTestId } = getRenderedComponent();
    const foundCustomHeaderComponent = getByTestId(testID);
    expect(foundCustomHeaderComponent).toBeTruthy();
  });
});
