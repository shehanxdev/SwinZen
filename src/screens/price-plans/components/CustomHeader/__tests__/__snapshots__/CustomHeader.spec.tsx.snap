// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomHeader Component should render CustomHeader component correctly 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginTop": 0,
      }
    }
    testID="CustomHeaderTestID"
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
          "paddingLeft": 16,
          "paddingRight": 16,
        }
      }
    >
      <SvgMock
        height={20}
        width={12}
      />
    </View>
    <Text
      style={
        [
          {
            "color": "#F8F9FA",
            "lineHeight": 28,
          },
          {
            "fontFamily": "SourceSansPro-SemiBold",
            "fontSize": 21,
            "fontWeight": "600",
            "textAlign": "center",
          },
        ]
      }
    >
      testTitle
    </Text>
    <View
      style={
        {
          "width": 48,
        }
      }
    />
  </View>
</View>
`;
